core/payment/src/dao/activity.rs

```diff
@@ -49,6 +56,50 @@ pub fn set_amount_accepted(
     agreement::increase_amount_accepted(&agreement_id, owner_id, &amount_delta, conn)
 }
 
+pub fn increase_amount_paid(
+    activity_id: &String,
+    owner_id: &NodeId,
+    amount: &BigDecimalField,
+    conn: &ConnType,
+) -> DbResult<()> {
+    assert!(amount > &BigDecimal::zero().into());
```

Comment:
What is more, it is a DAO module. IMHO DAO should be "stupid" - should enable only database operations without any logic.

Reply:
These assertions in DAO module are just sanity checks. They are for double-checking of **what should've been already validated** by other layers. Why did I put them if it's already being validated elsewhere? Because the payment is still under development and I'm not completely sure that I've implemented all the validation properly. These assertions are there to find bungs more easily. When the payment service is completed and thoroughly tested I'm happy to remove them.

I wish DAO was 'stupid' but it unfortunately it has to contain a little bit of logic because otherwise I cannot perform multiple database operations within a single transaction.

Reply:
I can add a comment like this to the assertions:
```
// TODO: Remove when payment service is production-ready.
```

Reply:
Ok, just leave comments :)